controllers/healthcheck_controller.go

```diff
@@ -170,68 +168,178 @@ func (r *HealthCheckReconciler) processHealthCheck(ctx context.Context, req ctrl
-			}
-			log.Info("Successfully Created", "ServiceAccount", servacc)
+		err := r.createRBACForWorkflow(log, healthCheck, "healthcheck")
+		if err != nil {
+			log.Error(err, "Error creating RBAC for HealthCheckWorkflow")
+			return ctrl.Result{}, err
+		}
+
+		log.Info("Creating Workflow", "namespace", wfNamespace, "generateNamePrefix", wfNamePrefix)
+		generatedWfName, err := r.createSubmitWorkflow(ctx, log, healthCheck)
+		if err != nil {
+			log.Error(err, "Error creating or submitting workflow")
+		}
+		r.watchWorkflowReschedule(ctx, log, wfNamespace, generatedWfName, healthCheck)
+	}
+	return ctrl.Result{}, nil
+}
 
-			clusrole, err := r.createClusterRole(r.kubeclient, amclusterrole)
+// SetupWithManager as used in main package by kubebuilder v2.0.0.alpha4
+func (r *HealthCheckReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	r.kubeclient = kubernetes.NewForConfigOrDie(mgr.GetConfig())
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&activemonitorv1alpha1.HealthCheck{}).
+		Complete(r)
+}
+
+func (r *HealthCheckReconciler) createRBACForWorkflow(log logr.Logger, hc *activemonitorv1alpha1.HealthCheck, workFlowType string) error {
+	level := hc.Spec.Level
+	hcSa := hc.Spec.Workflow.Resource.ServiceAccount
+	if hcSa == hc.Spec.RemedyWorkflow.Resource.ServiceAccount {
+		hc.Spec.RemedyWorkflow.Resource.ServiceAccount = hcSa + "-remedy"
+	}
+	remedySa := hc.Spec.RemedyWorkflow.Resource.ServiceAccount
+	wfNamespace := hc.Spec.RemedyWorkflow.Resource.Namespace
+	amclusterRole := hcSa + "-cluster-role"
+	amclusterRoleBinding := hcSa + "-cluster-role-binding"
+	amclusterRemedyRole := remedySa + "-cluster-role"
+	amclusterRoleRemedyBinding := remedySa + "-cluster-role-binding"
+	amnsRole := hcSa + "-ns-role"
+	amnsRoleBinding := hcSa + "-ns-role-binding"
+	amnsRemedyRole := remedySa + "-ns-role"
+	amnsRemedyRoleBinding := remedySa + "-ns-role-binding"
+
+	if workFlowType != "remedy" {
+		servacc, err := r.createServiceAccount(r.kubeclient, hcSa, wfNamespace)
+		if err != nil {
+			log.Error(err, "Error creating ServiceAccount for the workflow")
+		}
+		log.Info("Successfully Created", "ServiceAccount", servacc)
+	} else {
+		servacc1, err := r.createServiceAccount(r.kubeclient, remedySa, wfNamespace)
+		if err != nil {
+			log.Error(err, "Error creating ServiceAccount for the workflow")
+		}
+		log.Info("Successfully Created", "ServiceAccount", servacc1)
+	}
+	if level == "cluster" {
+
+		if workFlowType != "remedy" {
+			clusrole, err := r.createClusterRole(r.kubeclient, amclusterRole)
 			if err != nil {
-				log.Error(err, "Error creating ClusterRole for the workflow")
+				log.Error(err, "Error creating ClusterRole for the healthcheck workflow")
 			}
 			log.Info("Successfully Created", "ClusterRole", clusrole)
 
-			crb, err := r.CreateClusterRoleBinding(r.kubeclient, amclusterrolebinding, clusrole, sa, wfNamespace)
+			crb, err := r.createClusterRoleBinding(r.kubeclient, amclusterRoleBinding, amclusterRole, hcSa, wfNamespace)
```

Comment:
Same as the `serviceAccount`, if the role/clusterrole or the rolebinding/clusterrolebinding had errors during its creation, the HealthCheck should fail.